modules/speech-to-text-impl/src/main/java/org/opencastproject/speechtotext/impl/engine/WhisperEngine.java

```diff
@@ -209,7 +223,8 @@ public Map<String, Object> generateSubtitlesFile(File mediaFile,
       logger.debug("Transcription failed closing Whisper transcription process for: {}", mediaFile);
       throw new SpeechToTextEngineException(e);
     } finally {
-      IoSupport.closeQuietly(process);
+      IoSupport.closeQuietly(in);
```

Comment:
Can't you just use [Java's try-with-resources statement](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html). That's easier and while I'm not entirely sure, I think I remember that `IoSupport.closeQuietly` is deprecated.

Reply:
I used the try-with-resources statment for the BufferedReader, but it didn't work for `process`, because it doesn't implement `AutoClosable`. I used the `destroy()` and `destroyForcibly` methods instead. Is that okay?

Reply:
I agree with using try-with-resources wherever possible and with the BufferedReader. However, it can't be used with `transcriptionProcess`. Furthermore, I'd assume `IoSupport.closeQuietly` is fine to use, since it is part of our `common` module and not deprecated. Maybe you were thinking of another library Lars?